crates/factor-llm/src/lib.rs

```diff
@@ -0,0 +1,114 @@
+mod host;
+
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use spin_factors::{
+    ConfigureAppContext, Factor, InstanceBuilders, PrepareContext, RuntimeFactors,
+    SelfInstanceBuilder,
+};
+use spin_locked_app::MetadataKey;
+use spin_world::v1::llm::{self as v1};
+use spin_world::v2::llm::{self as v2};
+
+pub const ALLOWED_MODELS_KEY: MetadataKey<Vec<String>> = MetadataKey::new("ai_models");
+
+pub struct LlmFactor {
+    create_engine: Box<dyn Fn() -> Box<dyn LlmEngine> + Send + Sync>,
+}
+
+impl LlmFactor {
+    pub fn new<F>(create_engine: F) -> Self
+    where
+        F: Fn() -> Box<dyn LlmEngine> + Send + Sync + 'static,
+    {
+        Self {
+            create_engine: Box::new(create_engine),
+        }
+    }
+}
+
+impl Factor for LlmFactor {
+    type RuntimeConfig = ();
+    type AppState = AppState;
+    type InstanceBuilder = LlmDispatch;
```

Comment:
I will say I was confused for quite awhile reading this code what `LlmDispatch` was. I expected it to be called `InstanceState` like we do in some of the other factors. Maybe it would be nice to consistently name this? Maybe I just need to read the code better? Idk, 🤷 pretty ambivalent about what you choose to do here, but thought I would point out I struggled.

Reply:
I'm fine with renaming it to `InstanceState` - `LlmDispatch` was just used in the old `llm` implementation.